047-TrafficControlWithDapr/README.md

```diff
@@ -0,0 +1,73 @@
+# What The Hack - TrafficControlWithDapr
+
+## Introduction
+This repository contains several hands-on challenges that will introduce you to [Dapr](https://dapr.io/). You will start with a simple ASP.NET Core application that is composed of several microservices. In each challenge, you'll enhance the application by adding Dapr building blocks and components. At the same time, you'll configure the application to consume Azure-based backing services. When complete, you'll have implemented the following Dapr building blocks:
+
+- Service invocation
+- State-management
+- Publish / Subscribe
+- Bindings
+- Secrets management
+
+As Dapr can run on a variety of hosts, you'll start by running Dapr in self-hosted mode on your computer. Then, you'll deploy the Dapr application to run in Azure Kubernetes Service.
```

Comment:
You mention the attendee will run it locally, but all of the instructions I see refer to running it in Azure the entire time?  Did I miss something?

Reply:
you do both, you first build it locally, using local components (flat file for secrets, local SMTP server, local MQTT queue, etc), then you swap the Dapr components for the Azure-hosted ones, this is to demonstrate that Dapr abstracts your code from knowing the implementation details of how services are provided, you can just change the Dapr sidecar configuration to point to different backing implementations and your code never changes